gusto/time_discretisation/implicit_runge_kutta.py

```diff
@@ -91,31 +97,108 @@ def setup(self, equation, apply_bcs=True, *active_labels):
 
         super().setup(equation, apply_bcs, *active_labels)
 
-        self.k = [Function(self.fs) for i in range(self.nStages)]
+        if self.rk_formulation == RungeKuttaFormulation.predictor:
+            self.xs = [Function(self.fs) for _ in range(self.nStages)]
+        elif self.rk_formulation == RungeKuttaFormulation.increment:
+            self.k = [Function(self.fs) for _ in range(self.nStages)]
+        elif self.rk_formulation == RungeKuttaFormulation.linear:
+            raise NotImplementedError(
+                'Linear Implicit Runge-Kutta formulation is not implemented'
+            )
+        else:
+            raise NotImplementedError(
+                'Runge-Kutta formulation is not implemented'
+            )
 
     def lhs(self):
         return super().lhs
 
     def rhs(self):
         return super().rhs
 
-    def solver(self, stage):
-        residual = self.residual.label_map(
-            lambda t: t.has_label(time_derivative),
-            map_if_true=drop,
-            map_if_false=replace_subject(self.xnph, self.idx),
-        )
+    def res(self, stage):
+        """Set up the discretisation's residual for a given stage."""
```

Comment:
Can we make clear that this is only for the `predictor` formulation?

Reply:
Done